### DISCUSSION TITLE
Automatic conversion of column names from PascalCase to camelCase when working with Prisma, SQL and Nest.js

### DISCUSSION THREAD

**Discussion by itayperry** [prisma/prisma#20051] [Q&A]
HI there,
I have an SQL DB and when I fetch data (through Prisma) and the keys are PascalCase,
is there any way in Prisma to make all the data keys in camelCase after you fetch them?

I'd love for some help ♥

**Comment by nurul3101** [comment]
Hello @itayperry 👋 

You could use [Prisma Client Extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) to transform the data once it's fetched.
Here's an [example](https://github.com/prisma/prisma-client-extensions/tree/main/transformed-fields) that you can use as a reference.

The example demonstrates how to use a Prisma Client extension to transform result fields returned by queries. You could use the same approach to convert the results to camelCase.

**Reply by itayperry** [reply]
Hi @nurul3101, thank you for replying so fast ♥
I see examples for adding new keys and values for a specific table after fetching data from it, but I wish to do it for all the data in all the tables.. is there something more general?
Is there an `$extends` for **everything**?

**Reply by nurul3101** [reply]
> Is there an $extends for everything?

I believe you are looking for [$allOperations](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/query#modify-all-prisma-client-operations). Can you have a look?

**Reply by itayperry** [reply]
Hi @nurul3101, I'll look into that! Thank you so much!!

**Comment by janpio** [comment]
What do you refer to by "keys"? The field names in your models? Or some data that is being stored in your database?

**Comment by itayperry** [comment]
Hi there @janpio, I'm sorry for being unclear :) I think I meant the column names - they are PascalCase:
![aaa](https://github.com/prisma/prisma/assets/37377389/0cf2f3c7-00cd-4c0e-bb32-205935971ce2)


and I want them to be camelCase without adding `@map` to each field separately (I have a lot of tables and fields so it'll be hard 😅).
Is there any way to do it? And even if there isn't, is there some kind of "Prima middleware" where I can do it?
What I'm currently doing is just fetching the data and then using an `npm` package called `camelCase` with `JSON.parse()` and `JSON.stringify`. I just feel like I'm doing this the wrong way.

**Reply by janpio** [reply]
Unfortunately not, the Prisma schema represents the fields the way they should be returned by Prisma Client - and right now the only way to map that is explicitly with `@map`. We do not have any logic that could be applied to all fields automatically. 

So optimally you would configure that once to turn your database column names into the Prisma field names you want. I think you can use `F2` via our Prisma VSCode extension which might make it faster to do the renames. 

(If that does not work, I personally would probably write a script that parses the schema file and modifies each line individually [2 spaces at the beginning of a line is a good indicator that a field name follows, use everything until the next space, lowercase it, and then replace the existing value with `lowercase @map("original")` for example. That should help automate many of the necessary changes.])

**Reply by itayperry** [reply]
hi @janpio, I haven't heard of `F2`, I'll look into it!

> a script that parses the schema file and modifies each line individually

Any idea where I can find an example-script that I can start from? It doesn't have to do what I need to do... I just would love to see an example of it.

BTW, you've replied super fast! Thank you so much!

**Reply by janpio** [reply]
Unfortunately, I don't think we have an example of these scripts. Might be someone shared a snippet somewhere, but not sure.

(I would honestly start by asking ChatGPT how I can code what I described above as a shell script, or maybe a TS one off script. Then iterate from there. ChatGPT and other AI tools are _really_ got in turning simple instructions into one off scripts like that to manipulate simple text files.)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
